add static method to return match keys dict with custom property logic

mitchdawson1982 opened the pull request:
Add's a new static method to return matched fields with logic to handle fields within the custom property value.

Commit:
add static method to return match keys dict with custom property logic

Commit:
:octocat: Add Ingestion Syncer and Notifier repositories (#3638)

Add syncer and notifier repositories

Signed-off-by: Jacob Woffenden <jacob.woffenden@digital.justice.gov.uk>

Commit:
Update changelog and pyproject version

MatMoore commented on python-libraries/data-platform-catalogue/data_platform_catalogue/client/datahub/search.py:
```diff
@@ -114,6 +112,18 @@ def search(
+        for field in fields:
+            name = field.get("name")
+            value = field.get("value")
+            if name == "customProperties" and value != "":
```
Can you add a test for this case? I think the rest of this function is tests already but not the special case for customProperties

mitchdawson1982 replied:
Updated the query test case with a custom property match field.

dependabot[bot] committed:
pip(deps): bump boto3 from 1.34.53 to 1.34.57 in /containers/jml-extract-lambda/src/var/task (#3616)

* pip(deps): bump boto3 in /containers/jml-extract-lambda/src/var/task

Bumps [boto3](https://github.com/boto/boto3) from 1.34.53 to 1.34.57.
- [Release notes](https://github.com/boto/boto3/releases)
- [Changelog](https://github.com/boto/boto3/blob/develop/CHANGELOG.rst)
- [Commits](https://github.com/boto/boto3/compare/1.34.53...1.34.57)

---
updated-dependencies:
- dependency-name: boto3
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

* Bump botocore version, release version and changelog

---------

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: Michael Collins <15347726+michaeljcollinsuk@users.noreply.github.com>

Commit:
Update query test to include customProperties matched field

Commit:
Updated changlog with test details

mitchdawson1982 committed:
Merge branch 'main' into dp-3498-search-highlighting
